packages/core/test-support/helpers/contains-helpers.js

```diff
@@ -27,3 +27,62 @@ export function buttonContains(text) {
 export function linkContains(text) {
   return findByContains('a', text);
 }
+
+/**
+ * This function allows us to keep using the jquery notation for :contains without using jquery
+ *
+ * Example:
+ * Jquery: $(.class-thing:contains(blah) .metric .test:contains(other))
+ *
+ * findByContains:
+ * [...findByContains('.class-thing', 'blah').querySelectorAll('.metric .test')].find(el => el.innerText.trim().includes('other'))
+ *
+ * Non-Jquery Equivalent: [
+ *    ...[
+ *      ...document.querySelectorAll(
+ *        (.class-thing)
+ *      )].find(el => el.innerText.trim().includes('blah'))
+ *    .querySelectorAll('.metric .test')
+ *  ]
+ *  .find(el => el.innerText.trim().includes('other'))
+ *
+ * findContains: findContains(.class-thing:contains(blah) .metric .test:contains(other))
+ *
+ * @param {String} selector - jquery or native style selector
+ * @param {HTMLElement} [baseElement] - optional parameter to use as base element to select off of
+ * @returns {HTMLElement} - Element based on text it (or its parent) contains
+ */
+export function findContains(selector, baseElement) {
+  let nativeSelector,
+    parentElement = baseElement || document;
+
+  if (selector.includes(':contains(')) {
+    let splits = selector.split(':contains(');
+    let firstSelector = splits.shift(); //Selector without the :contains operator
+    let restOfSelector = splits.join(':contains('); //We only want to split on the first instance of :contains(, so join the rest back together if there were more
+    let endOfContentIndex = findEndOfContentIndex(restOfSelector);
```

Comment:
`/^(.*?):contains\((.+?)\)(.*)$/`

Group 1: firstSelector
Group 2: content
Group 3: nextSelector

Reply:
Considered using a regex for this and decided against it. Can't remember why though

Reply:
Now I remember, we select on some elements that have parentheses in them. The regex cuts off the content at the closing parenthesis. I figured it was safer to just find the content manually.

Reply:
The contains with parentheses are quoted right?

Reply:
`/^(.*?):contains\(\s*("(.*?)"|(.*?))\s*\)(.*)$/`

Group 1: firstSelector
Group 2:  Whatever is in the contains, trimmed
Group 3: If quoted, has text inside quote, otherwise undefined
Group 4: If not quoted, whatever is inside contains trimmed, otherwise undefined
Group 5: nextSelector

Reply:
okay, here we go.

`/^(.*?):contains\(\s*("(.*?)"|(.*?))\s*\)((:contains\(\s*(".*?"|.*?)\s*\)|\[.+?(=".*?"|.*?)?\]|[^\s+>~,])*)(.*)$/`

```perl
/^

# Lazy match everything before contains
# Group 1: firstSelectorPart1
(.*?)

# Match :contains()
# Group 2: Whatever is in the contains, trimmed
# Group 3: If quoted, has text inside quote, otherwise undefined
# Group 4: If not quoted, whatever is inside contains trimmed, otherwise undefined
:contains\(\s*("(.*?)"|(.*?))\s*\)

# Match trailing conjunctive selectors
# Group 5: firstSelectorPart2
((:contains\(\s*(".*?"|.*?)\s*\)|\[.+?(=".*?"|.*?)?\]|[^\s+>~,])*)

# Match everything after first selectors group
# Group 9: nextSelector
(.*)

$/
```

Now `firstSelector = firstSelectorPart1 + firstSelectorPart2`

Even this does not handle `,` selector operators properly.

Reply:
The contains with parentheses are not quoted, no

Reply:
Your regex still doesn't grab the closing parentheses correctly and cuts the content short when there's a `)` in it. 
<img width="1048" alt="Screen Shot 2019-03-31 at 2 18 25 PM" src="https://user-images.githubusercontent.com/23023478/55293820-199c2300-53c0-11e9-83a8-ffcbfc7d1c58.png">

Reply:
The idea is that parentheses need to be in quotes

Reply:
After all, there is no guaranty that the contains text will have balanced parens.

Reply:
I have no issue with using quotes, but I don't think adding quotes actually solves the problem, but rather changes it to a different problem. How would escaping the quotation marks change what the regex is parsing? It would still see it as a quote in the same way as if it was unescaped. 

Quotation syntax is optional in jquery and only used partly in navi so it shouldn't be a requirement here, but again, it's not the syntax that I mind. I just don't think it actually solves the problem it is meant to.

Reply:
The regex captures both with and without quotes.

Reply:
After taking a closer look at the jquery :contains behavior `/^(.*?):contains\((\s*"(.*?)"\s*|(.*?))\)(.*)$/` is a closer fit for their behavior.

It captures things like `:contains(skdjf " sldjf) -> 'skdjf " sldjf'` as expected, but can also properly capture `:contains("sldfjsd") -> 'sldfjsd'` as expected.

Reply:
jQuery does support escaping characters.  I don't think we need to support that functionality at this time.  Especially when we have a way of representing double quotes `:contains(")` and parens `:contains("()")`.

Reply:
It seems like the only difference in behavior with the jQuery capturing would be the quote escaping.  In jQuery they allow `:contains("text \" othertext") -> 'text " othertext'`.  They also allow `:contains('text " othertext') -> 'text " othertext'`.

Reply:
`/^(.*?):contains\((\s*"(.*?)"\s*|\s*'(.*?)'\s*|(.*?))\)(.*)$/` would grab single quote contents. 

And finally `/^(.*?):contains\((\s*"((?:\\.|.)*?)"\s*|\s*'((?:\\.|.)*?)'\s*|((?:\\.|.)*?))\)(.*)$/` grabs escaped characters the same way that jquery does.  This final method will require resolving the escaped characters in a second step.

Reply:
Here's a more compact version.  I've removed the white space trimming before and after quoted arguments because that is the jQuery behavior.

```javascript
/^(.*?):contains\((["']?)((?:\\.|.)*?)\2\)(.*)$/
```

Group 1: `firstSelector`
Group 3: `content` (without quotes)
Group 4: `nextSelector`